engine/src/main/java/com/redhat/cloud/notifications/templates/EmailTemplateService.java

```diff
@@ -16,15 +14,14 @@
     @Inject
     Engine engine;
 
-    public Uni<TemplateInstance> compileTemplate(String template, String name) {
-        return Uni.createFrom().item(engine.parse(template, null, name))
-                .onItem().transform(Template::instance);
+    public TemplateInstance compileTemplate(String template, String name) {
+        return engine.parse(template, null, name).instance();
     }
 
-    public Uni<String> renderTemplate(User user, Action action, TemplateInstance templateInstance) {
+    public String renderTemplate(User user, Action action, TemplateInstance templateInstance) {
```

Comment:
Wondering if (when we have the templates in the database) this compile/render thing should be a common thing across the backend/engine

Reply:
Maybe... we'll see when the DB thing is done, but I like the idea of having only one app (currently the `engine`) responsible for the templates rendering (separation of concerns).